casadm/cas_lib.c

```diff
@@ -2157,6 +2160,34 @@ static inline const char *partition_get_csv_col(CSVFILE *csv, int col,
 	return val;
 }
 
+static int calculate_max_allocation(uint16_t cache_id, const char *allocation,
+		uint32_t *part_size)
+{
+	int i;
+	float alloc = 0;
+	int res;
+
+	if (strnlen(allocation, MAX_STR_LEN) > 4)
+			return FAILURE;
+
+	for (i = 0; i < strnlen(allocation, MAX_STR_LEN); i++) {
+		if (!isdigit(allocation[i]) && allocation[i] != '.')
+			return FAILURE;
+	}
+
+	res = sscanf(allocation, "%f", &alloc);
+	if (res != 1)
+		return FAILURE;
```

Comment:
`strtof()` can do all of this and even more. E.g. `buf=".1."` will not be catched by this construction as an error, while using `strtof()` will prevent those kind of errors.